pkg/flowcli/project/transaction.go

```diff
@@ -0,0 +1,243 @@
+	return t.tx.AddArgument(arg)
+}
+
+func (t *Transaction) AddAuthorizers(addresses []string) error {
+	for _, address := range addresses {
+		err := t.AddAuthorizer(address)
+		if err != nil { // return error even if one breaks
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *Transaction) AddAuthorizer(address string) error {
+	authorizerAddress := flow.HexToAddress(address)
+	if authorizerAddress == flow.EmptyAddress {
+		return fmt.Errorf("invalid authorizer address provided %s", address)
+	}
+
+	t.tx.AddAuthorizer(authorizerAddress)
+	return nil
+}
+
+func (t *Transaction) SignerRole(role string) error {
+	t.role = signerRole(role)
+
+	switch t.role {
+	case SignerRoleAuthorizer: // Ignored if we're loading from a tx payload
+		err := t.AddAuthorizer(t.signer.Address().String())
+		if err != nil {
+			return err
+		}
+	case SignerRolePayer:
+		if t.payer != t.signer.Address() {
+			return fmt.Errorf("role specified as Payer, but Payer address also provided, and different: %s != %s", t.payer, t.signer.Address())
+		}
+	case SignerRoleProposer: // Just sign payload, no special actions needed
+	default:
+		return fmt.Errorf("unknown role %s", role)
+	}
+
+	return nil
+}
+
+func (t *Transaction) Sign() (*Transaction, error) {
+	keyIndex := t.signer.DefaultKey().Index()
+	signerAddress := t.signer.Address()
+	signer, err := t.signer.DefaultKey().Signer(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	switch t.role {
+	case SignerRoleAuthorizer, SignerRoleProposer:
+		err := t.tx.SignPayload(signerAddress, keyIndex, signer)
+		if err != nil {
+			return nil, fmt.Errorf("failed to sign transaction: %s", err)
+		}
+	case SignerRolePayer:
+		err := t.tx.SignEnvelope(signerAddress, keyIndex, signer)
+		if err != nil {
+			return nil, fmt.Errorf("failed to sign transaction: %s", err)
+		}
+	}
+
+	return t, nil
+}
+
+func (t *Transaction) AddContractsFromArgs(contractArgs []string) error {
+	for _, contract := range contractArgs {
+		contractFlagContent := strings.SplitN(contract, ":", 2)
+		if len(contractFlagContent) != 2 {
+			return fmt.Errorf("wrong format for contract. Correct format is name:path, but got: %s", contract)
+		}
+		contractName := contractFlagContent[0]
+		contractPath := contractFlagContent[1]
+
+		contractSource, err := util.LoadFile(contractPath)
+		if err != nil {
+			return err
+		}
+
+		t.AddContract(contractName, string(contractSource))
+	}
+
+	return nil
+}
+
+func (t *Transaction) AddContract(name string, source string) {
+	t.contracts = append(t.contracts,
+		templates.Contract{
+			Name:   name,
+			Source: source,
+		},
+	)
+}
+
+func (t *Transaction) SetCreateAccount(keys []*flow.AccountKey) {
+	t.tx = templates.CreateAccount(keys, t.contracts, t.signer.Address())
+}
+
+func (t *Transaction) SetUpdateContract(name string, source string) {
+	t.AddContract(name, source)
+	t.tx = templates.UpdateAccountContract(t.signer.Address(), t.contracts[0])
```

Comment:
Instead of doing `t.contracts[0]`, I would just do:

```go
contract := templates.Contract{
  Name: name,
  Source: source,
}

t.tx = templates.UpdateAccountContract(t.signer.Address(), contract)